From 8fdb07bca333237082d6bc377e5c8623aa9cff3a Mon Sep 17 00:00:00 2001
From: Huacai Chen <chenhuacai@loongson.cn>
Date: Mon, 11 Nov 2024 21:21:49 +0800
Subject: [PATCH 018/280] FROMLIST: drm: Remove redundant statement in
 drm_crtc_helper_set_mode()

Commit dbbfaf5f2641a ("drm: Remove bridge support from legacy helpers")
removes the drm_bridge_mode_fixup() call in drm_crtc_helper_set_mode(),
which makes the subsequent "encoder_funcs = encoder->helper_private" be
redundant, so remove it.

Cc: stable@vger.kernel.org
Fixes: dbbfaf5f2641a ("drm: Remove bridge support from legacy helpers")
Signed-off-by: Huacai Chen <chenhuacai@loongson.cn>
Reviewed-by: Jani Nikula <jani.nikula@intel.com>

Link: https://lore.kernel.org/all/20241111132149.1113736-1-chenhuacai@loongson.cn/
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/gpu/drm/drm_crtc_helper.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/gpu/drm/drm_crtc_helper.c b/drivers/gpu/drm/drm_crtc_helper.c
index 0955f1c385dd..39497493f74c 100644
--- a/drivers/gpu/drm/drm_crtc_helper.c
+++ b/drivers/gpu/drm/drm_crtc_helper.c
@@ -334,7 +334,6 @@ bool drm_crtc_helper_set_mode(struct drm_crtc *crtc,
 		if (!encoder_funcs)
 			continue;
 
-		encoder_funcs = encoder->helper_private;
 		if (encoder_funcs->mode_fixup) {
 			if (!(ret = encoder_funcs->mode_fixup(encoder, mode,
 							      adjusted_mode))) {
-- 
2.48.1

